tests/drivers/build_all/sensors_i_z.conf

```diff
@@ -1,5 +1,6 @@
 CONFIG_TEST=y
 CONFIG_TEST_USERSPACE=y
+CONFIG_NEWLIB_LIBC=y
```

Comment:
what's going on here?

Reply:
STM hal is including math.h and minimal libc does not provided math.h

Reply:
I don't remember this change when I approved the PR.  I'd rather see minimal libc gain math.h that the HALs need, assuming it's not going to involve anything beyond the header.

Reply:
@pabigot 
>  I'd rather see minimal libc gain math.h that the HALs need, assuming it's not going to involve anything beyond the header.

Actually the HAL kconfig `HAS_STMEMSC` was hiding this  by using `imply NEWLIB_LIBC`.

Reply:
Hmpf.  Still, doesn't this change the behavior for all platforms?

Reply:
> Still, doesn't this change the behavior for all platforms?

Actually this PR does not change any behavior, new libc was already required but it was hidden from the `prj.conf` because it was selected (via `imply`) by `HAS_STMEMSC` by changing this `select` in a `depend` it becomes visible in the `pr.conf` file

Reply:
I guess this is more test infrastructure magic, because I only see five sensors that select `HAS_STMEMSC` so I would expect a behavior change in every other sensor that wouldn't normally be building with newlib.

I tried `sanitycheck -T samples/sensor` on a master branch with `Kconfig.st`'s newlib dependency removed and everything passed except `qemu_x86_long samples/sensor/thermometer/sample.sensor.thermometer` which failed because the test's exclude clause is incorrect (what is going on with CI these days, anyway?).

So I can't see how to confirm or deny that this is necessary, or whether there's another workaround.  I'll leave it to somebody else.

Reply:
Thanks.  Those also pass on master with the `imply NEWLIB_LIBC` removed.

Reply:
Hmm strange, just double checked myself against latest master (c2869c7e78cff4c643153a4f1b11971189fffa8b) with the `imply NEWLIB_LIBC` removed from  `config HAS_STMEMSC` and if I run `sanitycheck -N -i -T zephyr/samples/shields/x_nucleo_iks01a3` 2 out of 2 tests fails due to missing `math.h`

```
.../modules/hal/st/sensor/stmemsc/lis2dw12_STdC/driver/lis2dw12_reg.h:48:10: fatal error: math.h: No such file or directory
 #include <math.h>
          ^~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

Reply:
I have no explanation:
```
tirzah[281]$ sanitycheck -N -i -T samples/shields/x_nucleo_iks01a3
JOBS: 24
Cleaning output directory /mnt/nordic/zp/zephyr/sanity-out
Selecting default platforms per test case
Building testcase defconfigs...
Filtering test cases...
2 tests selected, 388 tests discarded due to filters
total complete:    2/   2  100%  failed:    0
2 of 2 tests passed with 0 warnings in 18 seconds
tirzah[282]$ git diff | cat
diff --git a/modules/Kconfig.st b/modules/Kconfig.st
index f2827fc548..7d5b8dd4d6 100644
--- a/modules/Kconfig.st
+++ b/modules/Kconfig.st
@@ -7,8 +7,6 @@ config HAS_STLIB
 
 config HAS_STMEMSC
 	bool
-	imply NEWLIB_LIBC
-	imply NEWLIB_LIBC_FLOAT_PRINTF
 
 if HAS_STMEMSC
 
tirzah[283]$ git one | head -1
* c2869c7e78 (HEAD -> upstream/master, r.upstream/master) [Mon Sep 16 20:19:58 2019 +0200] doc: reference: settings: list non-volatile storage backend
tirzah[284]$ 
```
but if I go into `sanity-out/frdm_k64f/samples/shields/x_nucleo_iks01a3/standard/test` and `ninja menuconfig` to disable `NEWLIB_LIBC` then I can see the problem.  So something unknown is still introducing the dependency.

However, at least I can reproduce it.  The following seems to be sufficient for the case I can reproduce:
```
#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_MATH_H_
#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_MATH_H_

/* STMEMSC drivers require this header with this typedef. */
typedef float float_t;

#endif /*  ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_MATH_H_ */
```